coq.opam

```diff
@@ -1,18 +1,29 @@
-opam-version: "1.2"
+synopsis: "The Coq Proof Assistant"
+description: """
+Coq is a formal proof management system. It provides
+a formal language to write mathematical definitions, executable
+algorithms and theorems together with an environment for
+semi-interactive development of machine-checked proofs. Typical
+applications include the certification of properties of programming
+languages (e.g. the CompCert compiler certification project, or the
+Bedrock verified low-level programming library), the formalization of
+mathematics (e.g. the full formalization of the Feit-Thompson theorem
+or homotopy type theory) and teaching.
+"""
+opam-version: "2.0"
 maintainer: "The Coq development team <coqdev@inria.fr>"
 authors: "The Coq development team, INRIA, CNRS, and contributors."
 homepage: "https://coq.inria.fr/"
 bug-reports: "https://github.com/coq/coq/issues"
-dev-repo: "https://github.com/coq/coq.git"
+dev-repo: "git+https://github.com/coq/coq.git"
 license: "LGPL-2.1"
-
-available: [ ocaml-version >= "4.05.0" ]
+doc: "https://coq.github.io/coq/api"
```

Comment:
this URL is a 404

Reply:
This is required by the `topkg` / `dune-release` linter, it is not used in practice unless you do `dune-release publish doc`.

Dunno what to do about it.

Reply:
I guess I will report a bug. If you modify it this happens:
```
$ dune-release distrib
Wrote archive _build/coq-V8.10+alpha-647-g5efe3fea6c.tbz
[ OK ] lint opam file coq.opam.
dune-release: [ERROR] Could not derive publication directory $PATH from opam
                      doc field value "https://coq.inria.fr/master/api";
                      expected the pattern
                      $SCHEME://$USER.github.io/$REPO/$PATH
[FAIL] lint _build/coq-V8.10+alpha-647-g5efe3fea6c failure: 1 errors.
```

Reply:
LOL that's a pretty aggressive expectation indeed.

Reply:
> Dunno what to do about it.

I don't know what dune-release is. But obliging people to put a dummy URL is silly.
Unless there is something like `--warn=-XX`, you should report a feature whish upstream.

Reply:
> Unless there is something like `--warn=-XX`, you should report a feature whish upstream.

We can disable the linter but indeed it should be improved upstream. Dune release is the dune-aware for of `topkg`.

Reply:
Is not so silly when you consider the `topkg` release workflow; but indeed I do agree it has to be tweaked.

Reply:
Anyways this should have minimal impact for Coq as of today as we are not going to use the `publish` command for now.

Reply:
Disabling the linter seems like the thing to do until the bug is fixed, yes.